src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala

```diff
@@ -163,6 +168,72 @@ class HudiOutputWriter(props: Map[String, Object], hudiOutput: Option[Hudi]) ext
       case None =>
     }
   }
-  // scalastyle:on cyclomatic.complexity
-  // scalastyle:on method.length
+
+
+  private def updateSchema(dataFrame: DataFrame): DataFrame = {
+    var df = dataFrame
+    val ss = dataFrame.sparkSession
+    var fieldMap = Map[String, DataType]()
+
+    // If table existed before, get it's schema
+    val previousSchema: Option[StructType] = this.hudiOutputProperties.tableName match {
+      case Some(tableName) => {
+        ss.catalog.tableExists(tableName) match {
+          case true => Option(ss.table(tableName).schema)
+          case false => None
+        }
+      }
+      case None => None
+    }
+
+    // Update the schema
+    val schema = StructType(df.schema.fields.flatMap(
+      field => {
+        // Add nullability, not on by default
+        val newField = field.copy(nullable = true)
+        val fieldName = newField.name
+        var returnedFields = List[StructField]()
+
+        // By default we will remove completely null columns (will return them if existed in previous schemas), you can disable this behaviour
+        this.hudiOutputProperties.dropNullColumns match {
+          case Some(true) => {
+            // Column is detected as having only null values, we need to remove it
+            isOnlyNullColumn(df, fieldName) match {
+              case true => {
+                df = df.drop(fieldName)
+
+                // Check if removed column existed in a previous schema, if so, use the previous schema definition
```

Comment:
Why does it matter?

Reply:
In case this column was once written to hive (with actual data), and now the column is completely null, we will drop it and add it to the DF with its previous schema definition. 
Otherwise, Hudi will fail to write the parquet, due to schema mismatch.

Reply:
Got it. In case it had previous values they should be represented in the schema with the right data type.